spyder/plugins/editor/tests/test_introspection.py

```diff
@@ -109,8 +109,23 @@ def test_hide_widget_completion(setup_editor):
     # Set cursor to start
     code_editor.go_to_line(1)
 
-    qtbot.keyClicks(code_editor, 'from numpy [')
-    assert completion.isVisible() is False
+    # Complete from numpy --> from numpy import
+    qtbot.keyClicks(code_editor, 'from numpy ')
+    qtbot.wait(20000)
+
+    # press tab and get completions
+    with qtbot.waitSignal(completion.sig_show_completions,
+                          timeout=10000) as sig:
+        qtbot.keyPress(code_editor, Qt.Key_Tab)
+
+    # Check the completion widget is visible
+    assert completion.isHidden() is False
+
+    # Write a delimeter on the code editor
+    qtbot.keyClicks(code_editor, '(')
+
+    # Check the completion widget is not visible
+    assert completion.isHidden() is True
```

Comment:
@steff456 could you rewrite this test so that the test is run for all the chars?

Perhaps using a `for` and iterating over `self.BRACKETS_CHAR` so that we are indeed checking for all delimiters?

Reply:
Introspection tests take a lot of time to run, so I don't think rerunning this test for all bracket chars would be necessary.

Reply:
At least then it should randomly pick the char from the list so we are randomly testing all the options, otherwise this test is broken and incomplete.

Reply:
Something like

```python
import random
delimeter = random.choice(BRACKETS_CHAR)
```

Reply:
Ok, that's better. @steff456, please implement Gonzalo's suggestion of picking up a random bracket char and running the test with it.

Reply:
Ok, I will add all delimiters in the test

Reply:
There's no need to add all delimiters, just one selected at random.